File: VenueBitApp/VenueBitApp/Core/Networking/Models/Order.swift
import Foundation

struct Order: Codable, Identifiable {
    let id: String
    let userId: String
    let status: OrderStatus
    let items: [CartItem]
    let subtotal: Double
    let fees: Double
    let total: Double
    let tickets: [Ticket]
    let createdAt: String
}

enum OrderStatus: String, Codable {
    case pending
    case confirmed
    case cancelled
    case refunded
}

struct CheckoutRequest: Codable {
    let cartId: String
    let userId: String
    let payment: PaymentInfo
}

struct PaymentInfo: Codable {
    let cardLast4: String
}

struct OrderResponse: Codable {
    let data: Order
}

struct OrdersListResponse: Codable {
    let data: [Order]
}

File: webapp-desktop/src/types/order.ts
export interface Order {
  id: string;
  userId: string;
  items: OrderItem[];
  subtotal: number;
  serviceFee: number;
  processingFee?: number;
  total: number;
  status: 'pending' | 'confirmed' | 'completed' | 'cancelled' | 'refunded';
  createdAt: string;
  payment?: {
    cardLast4: string;
  };
}

export interface OrderItem {
  id: string;
  eventId: string;
  eventTitle: string;
  eventDate: string;
  eventTime: string;
  venueName: string;
  seats: OrderSeat[];
  subtotal: number;
}

export interface OrderSeat {
  id: string;
  section: string;
  row: string;
  seatNumber: number;
  price: number;
}

export interface CheckoutRequest {
  cartId: string;
  userId: string;
  cart: import('./cart').Cart;
  payment: {
    cardLast4: string;
    cardholderName?: string;
  };
}
